bpfprogs/bpf.go

```diff
@@ -452,23 +452,20 @@ func (b *BPF) Start(ifaceName, direction string, chain bool) error {
 
 // UpdateBPFMaps - Update the config ebpf maps via map arguments
 func (b *BPF) UpdateBPFMaps(ifaceName, direction string) error {
-	for k, val := range b.Program.MapArgs {
-
-		if v, ok := val.(string); !ok {
-			err := fmt.Errorf("update map args is not a string for the ebpf program %s", b.Program.Name)
-			log.Error().Err(err).Msgf("failed to convert map args value into string for program %s", b.Program.Name)
-			return err
-		} else {
-			log.Info().Msgf("Update map args key %s val %s", k, v)
-
-			bpfMap, ok := b.BpfMaps[k]
-			if !ok {
-				if err := b.AddBPFMap(k); err != nil {
-					return err
-				}
-				bpfMap = b.BpfMaps[k]
+	for _, val := range b.Program.MapArgs {
+		bpfMap, ok := b.BpfMaps[val.Name]
+		if !ok {
+			if err := b.AddBPFMap(val.Name); err != nil {
+				return err
 			}
-			bpfMap.Update(v)
+			bpfMap = b.BpfMaps[val.Name]
+		}
+		if err := bpfMap.DeleteAllEntries(); err != nil {
```

Comment:
Can we improvise the logic here, first update existing and add new values in the map args. The last step is removing missing values. It will be less disruptive.

Reply:
Done